webapi/src/main/scala/org/knora/webapi/responders/admin/UsersResponderADM.scala

```diff
@@ -983,6 +997,22 @@ class UsersResponderADM(responderData: ResponderData) extends Responder(responde
         _              = if (!projectExists) throw NotFoundException(s"The project $projectIri does not exist.")
 
         // get users current project membership list
+        currentProjectMemberships <- userProjectMembershipsGetADM(
+                                       userIri = userIri,
+                                       requestingUser = KnoraSystemInstances.Users.SystemUser
+                                     )
+
+        currentProjectMembershipIris = currentProjectMemberships.map(_.id)
+
+        // check if user is already project member and if not throw exception
+
+        _ = if (!currentProjectMembershipIris.contains(projectIri)) {
+              throw BadRequestException(
+                s"User $userIri is not a member of project $projectIri. A user needs to be a member of the project to be added as project admin."
+              )
+            }
```

Comment:
in terms of usability of our system, might it make more sense to just add the user as a project member when they are made project admin?

Reply:
In order to decide what is more user friendly, I think we need to have a proper domain model to see what the possible user flows are to make a user a member or admin of a project. Without it, everything and nothing makes sense.. :)

Reply:
As discussed: this follows the logic of the current implementation of dsp-app. First, the user is added as member. Afterwards he can be added as project admin. I leave it like this.